src/components/views/rooms/RoomList.tsx

```diff
@@ -289,12 +285,11 @@ export default class RoomList extends React.PureComponent<IProps, IState> {
         // shallow-copy from the template as we need to make modifications to it
         this.tagAesthetics = objectShallowClone(TAG_AESTHETICS);
         this.updateDmAddRoomAction();
-
-        this.dispatcherRef = defaultDispatcher.register(this.onAction);
-        this.roomStoreToken = RoomViewStore.addListener(this.onRoomViewStoreUpdate);
     }
 
     public componentDidMount(): void {
+        this.dispatcherRef = defaultDispatcher.register(this.onAction);
+        this.roomStoreToken = RoomViewStore.addListener(this.onRoomViewStoreUpdate);
```

Comment:
is it actually more performant to move things to `componentDidMount`? Over the last couple years we've started using the constructor more and more, slowly getting rid of `componentDidMount` - putting aside arguments about whether that is correct behaviour, I'm curious if there's a performance impact.

Reply:
It depends, generally speaking there can be small performance improvements when the component is rendered but not mounted 

But in my opinion the real power of this is that you colocate all your event listeners, if you have some event listeners that you need to place on DOM elements, you will have to perform that in `componentDidMount`. And on top of that if it is immediatly followed by `componentWillUnmount` you're one step closer to remembering to clean up your listeners

It also helps in tests where you shallow render your components

Overall moving away from lifecycles makes sense, this is pretty much what the React team has been advocating for, but their solution seems to be using hooks instead